### ISSUE TITLE
garden + blog question

### ISSUE THREAD

**Issue by ShannaSeigel** [maximevaillancourt/digital-garden-jekyll-template#19]
Hi there,

I'm working on hosting my notes online and have been looking at themes I can use with github pages. I really like your digital garden style! I don't have a background in programming so there's a lot I'm trying to learn. If I'm using your digital garden would I also be able to create blog posts? From looking at your set up guide I'm guessing hosting a blog would be done separately?

I hope this makes sense!

**Comment by maximevaillancourt** [comment]
Hi Shanna!

You could definitely use the template as a starting point, but as is, the template doesn't offer a traditional timeline view of all notes by chronological order, so it may not meet your needs.

You may instead prefer to draw inspiration from my [personal website](https://maximevaillancourt.com/) ([view repository on GitHub](https://github.com/maximevaillancourt/maximevaillancourt.com)), which is a combination of a blog and a digital garden split into different sections. Feel free to fork both this template's repository and my personal website's repository, and play with each to see which one fits your needs. :slightly_smiling_face:

**Comment by ShannaSeigel** [comment]
Thanks for the fast reply! I completely missed the graph and list of notes on your personal website the first time through. I will definitely start exploring these options!

**Comment by Am3ra** [comment]
@maximevaillancourt , I believe theres a jekyll plugin or similar to generate a list of last-modified files/posts

**Comment by maximevaillancourt** [comment]
Closing this, feel free to reopen if needed :slightly_smiling_face:

**Comment by ShannaSeigel** [comment]
Hey again!

So, I have no idea what I'm doing and my only experience with the command line is imitating what other people do in tutorials (just as an explanation if I'm missing something obvious). I was able to install/update Ruby, RubyGems, and Git. Then I followed your directions for cloning the directory and installing "bundle." The problem is when I get to "bundle exec jekyll serve" and I get an error. 

Maybe this is super basic but are you able to tell what I'm doing wrong? It looks like something is missing?

![Screen Shot 2020-11-03 at 18 48 14](https://user-images.githubusercontent.com/31891821/98053061-c133cf80-1e05-11eb-915a-d7734540fd64.png)

**Comment by maximevaillancourt** [comment]
You're almost there! It looks like you need to install Jekyll. Try doing `gem install jekyll`, then run `bundle exec jekyll serve` again. Let me know if you get stuck 🙂

**Comment by ShannaSeigel** [comment]
🤦🏻‍♀️ Definitely did not consider that. I'm still get an error unfortunately. Now it's that I don't have write permissions for "Library/Ruby/Gems/2.6.0." Though when I updated earlier I was seeing version 2.7.2.

**Comment by maximevaillancourt** [comment]
It sounds like there are two things at play:
- Write permissions issues are expected since we're essentially trying to installing a gem globally. Using `sudo gem install jekyll` will probably fix this by granting additional permissions to bundle to install the gem in system directories ([more information about sudo](https://en.wikipedia.org/wiki/Sudo)).
- There are multiple versions of Ruby installed on your machine. You may want to look into installing Ruby Version Manager (RVM) to manage Ruby versions, but this shouldn't be necessary to start using the digital garden template. Of course, the more recent, the better: if you can use Ruby 2.7.2, do that! :slightly_smiling_face:

**Comment by ShannaSeigel** [comment]
Thank you again for such fast replies! I still got an error with `sudo gem install jekyll`
I tried checking write permissions for different folders but that didn't change anything. I'll try looking at the version manager next.

**Comment by maximevaillancourt** [comment]
Using RVM could potentially solve this indeed, since gems would be installed in your home directory instead of system directories. :slightly_smiling_face:

**Comment by ShannaSeigel** [comment]
Let me know if I should open a separate issue!

After a lot of trial and error I still haven't gotten jekyll to install. I'll worry about this again when I have some more downtime. I do have a question though. Going through the process of downloading all these programs is just to see the site locally before publishing it online? What if I didn't care about seeing the page locally and would look at the page live through GitHub pages instead? Is that possible?

I tried just cloning the digital garden repo and turning on github pages. The site would never build, I kept getting an error: 

> `/_layouts/notes.html` that is a symlink or does not exist in your `_includes` directory

I tried copying that file to "_includes" with no difference. I deleted the file and got pages to load but then of course the notes weren't formatted.

Basically, I'm trying to figure out a way to use your template for my garden notes without a lot of website design/building knowledge because there's still a lot for me to learn. I love the graph design you've created along with the backlinks/related notes so I'm hoping to figure this out!

**Comment by maximevaillancourt** [comment]
> After a lot of trial and error I still haven't gotten jekyll to install. I'll worry about this again when I have some more downtime.

Sorry to read that. Hopefully, using RVM and using a different Ruby version may fix this. What's the specific error you get when it fails?

> Going through the process of downloading all these programs is just to see the site locally before publishing it online? What if I didn't care about seeing the page locally and would look at the page live through GitHub pages instead? Is that possible?

You could do _almost_ exactly that, but not with GitHub Pages because it doesn't support custom Jekyll plugins (I had to write a [custom plugin to have the backlinks/related notes feature](https://github.com/maximevaillancourt/digital-garden-jekyll-template/blob/f297a68cbf517ba145e35f447f90799212610f14/_plugins/bidirectional_links_generator.rb)). Netlify supports custom Jekyll plugins, so that's why I suggest using their service in the [guide](https://maximevaillancourt.com/blog/setting-up-your-own-digital-garden-with-jekyll).

**Comment by ShannaSeigel** [comment]
I was still running into a version issue. Even after specifying 2.7.2 I was seeing "no write permission for 2.6.x" I think I will try working with Netlify in the meantime and see how that goes!
